sympy/polys/rootoftools.py

```diff
@@ -658,49 +698,139 @@ def _eval_evalf(self, prec):
-                    if self.is_real:
-                        if (a <= root <= b):
+                    if self.is_real or self.is_imaginary:
+                        if not bool(root.imag) == self.is_real and (
+                                a <= root <= b):
+                            if self.is_imaginary:
+                                root = mpc(mpf('0'), root.real)
                             break
                     elif (ax <= root.real <= bx and ay <= root.imag <= by):
                         break
                 except (UnboundLocalError, ValueError):
                     pass
                 interval = interval.refine()
 
-        return (Float._new(root.real._mpf_, prec)
-                + I*Float._new(root.imag._mpf_, prec))
+        # update the interval so we at least (for this precision or
+        # less) don't have much work to do to recompute the root
+        self._set_interval(interval)
+        return (Float._new(root.real._mpf_, prec) +
+            I*Float._new(root.imag._mpf_, prec))
 
-    def eval_rational(self, tol):
-        """
-        Return a Rational approximation to ``self`` with the tolerance ``tol``.
+    def _eval_evalf(self, prec, **kwargs):
+        """Evaluate this complex root to the given precision.
+
+        Examples
+        ========
+
+        >>> from sympy import rootof
+        >>> from sympy.abc import x
+        >>> r = rootof(4*x**5 + 16*x**3 + 12*x**2 + 7, 0)
+        >>> interval = r._get_interval()
+        >>> r.evalf(2)
+        -0.98
 
-        This method uses bisection, which is very robust and it will always
-        converge. The returned Rational instance will be at most 'tol' from the
-        exact root.
+        Notes
+        =====
 
-        The following example first obtains Rational approximation to 1e-7
-        accuracy for all roots of the 4-th order Legendre polynomial, and then
-        evaluates it to 5 decimal digits (so all digits will be correct
-        including rounding):
+        The interval is refined until the width of the root bounds
+        guarantees that any decimal approximation within those bounds
+        will satisfy the desired precision.
+
+        A side effect of this routine is that the refined interval is
+        stored so subsequent requests at or below the requested
+        precision will not have to recompute the root bounds and will
+        return very quickly.
+
+        >>> print('interval was %s and is now %s' % (interval, r._get_interval()))
+        interval was (-1, 0) and is now (-165/169, -206/211)
+
+        The `eval_approx` method will also find the root to a given
+        precision but the interval is not modified unless the search
+        for the root fails to converge within the root bounds.
+        """
+        # all kwargs are ignored
+        return self.eval_rational(n=prec_to_dps(prec))._evalf(prec)
+
+    def eval_rational(self, dx=None, dy=None, n=15):
```

Comment:
This now works for non-real roots and takes absolute (dx, dy) error bounds or a precision/relative bound (n).